src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -129,6 +129,31 @@ private static bool IsExcludedFromCodeCoverage(MethodSymbol method)
             return false;
         }
 
+        private BoundExpression GetFileIndices()
+        {
+            var documents = PooledHashSet<DebugSourceDocument>.GetInstance();
```

Comment:
>DebugSourceDocument [](start = 42, length = 19)

Would it be better to store indices instead? They are integers, the size is smaller, comparison is faster.